test/serializer/abstract/Throw8.js

```diff
@@ -0,0 +1,34 @@
+function call(fn) {
+  var template = {};
+  if (global.__makeSimple) __makeSimple(template);
+  function residualCall(fn) {
+    var value;
+    var exception;
+    var success = true;
+    try {
+      value = fn();
+    } catch (e) {
+      exception = e;
+      success = false;
+    }
+    return {value, exception, success};
+  }
+  var res = global.__residual? __residual(template, residualCall, fn) : residualCall(fn);
+  if (!res.success) {
+    throw res.exception;
+  }
+  return res.value;
+}
+
+var fn = global.__abstract ? __abstract('function', '(function () { })') : function () { };
+
+var x = 1;
+try {
+  call(fn);
+  x = 2;
+  //call(fn);
```

Comment:
Is this why this is only a partial fix?

Reply:
Yes, the second call happens in a forked path. Merging the forked path that results from the call with the forked path doing the call, while keeping the interpreter in a sane state does not work yet.